csharp/ql/lib/semmle/code/csharp/dataflow/FlowSummary.qll

```diff
@@ -168,7 +168,8 @@ private SummaryComponent delegateSelf() {
 
 private predicate mayInvokeCallback(Callable c, int n) {
   c.getParameter(n).getType() instanceof SystemLinqExpressions::DelegateExtType and
-  not c.fromSource()
+  not c.hasBody() and
+  (if c instanceof AddEventAccessor then not c.fromSource() else any())
```

Comment:
This same pattern is also used in `SsaImpl.qll`. Should it be refactored into a predicate?